Add TypeScript definition

fregante opened the pull request:
@sindresorhus pls cek

<img src="https://i.imgur.com/AfgkjCf.jpg">

`tsc` passes, but does that mean anything?

fregante committed:
Linting

fregante committed:
Linting

fregante committed:
Tentatively add types

fregante committed:
Add support for NodeList and DocumentFragment

fregante committed:
Use standard terms: selector, baseElement

fregante committed:
4.2.0-0

fregante committed:
Merge branch 'master' into types

fregante commented:
For reference, here's a TypeScript issue about automatically determining the element from a compound selector (plain element selectors already work correctly!) https://github.com/Microsoft/TypeScript/issues/29037

sindresorhus commented on index.d.ts:
```diff
@@ -0,0 +1,42 @@
+
+type baseElements = Element | Element[] | NodeList | DocumentFragment;
+
+export interface SelectDom {
```
I think you want to "export default" this one, so it's imported when you import `select-dom`.

Relevant: https://github.com/Microsoft/TypeScript/issues/3792#issuecomment-303526468

sindresorhus commented on index.d.ts:
```diff
@@ -0,0 +1,42 @@
+// https://github.com/Microsoft/TypeScript/blob/9d3707d671592d030386956c9ce39e539b8d0972/src/lib/dom.generated.d.ts#L10581
+
+type baseElements = Element | Element[] | NodeList | DocumentFragment;
```
`baseElements` => `BaseElements`

sindresorhus commented:
You should write some type tests: https://github.com/sindresorhus/typescript-definition-style-guide#testing

fregante committed:
Fix export

fregante committed:
4.2.0-1

fregante committed:
Follow suggestions by @nickytonline

https://github.com/sindresorhus/refined-github/pull/1750#issuecomment-460516305

fregante committed:
4.2.0-2

fregante commented:
Does tsd-check work in the browser though?

sindresorhus commented:
@bfred-it No. Why would you need it to?

fregante commented:
Because the other regular tests are run in electron. Select-dom calls `document.querySelector`

fregante committed:
Add support for DocumentFragment on `select()`

fregante committed:
4.2.0-3

sindresorhus commented:
@bfred-it `tsd-check` doesn't actually evaluate the file. It just runs the TS type checker on it.

fregante commented:
@sindresorhus I'll add that in the future. Releasing this for now since it seems to be working on RGH already
